### PR TITLE
Create march-03.md

### PR DESCRIPTION
**Pull request by locks**

### CODE REVIEW

File: ember.js/march/march-03.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+YK: It sounds like you're saying are there a few different cases that cause eager reification that you're addressing.
+
+RJ: Right, I just thought there was a single thing triggering a performance cliff, so my mental model was wrong.
+
+RJ: So my next question is something not discussed in the router service RFC. Right now to generate a URL we have to look up all of the routes, their parent routes, etc. That shouldn't be surprising to anyone.
+
+YK: It sounds like you're leading us towards the "separate URL serialization out from `Route`" discussion that we have discussed in the past.
+
+RJ: Currently we go and grab the route (Kris, even in engines we do this work still—once an engine is laoded anyway). I would like to lump the `serializeQueryParam` in with the main `serialize` method. Query param serialization is very odd right now. All query params serialization happens in Ember while all other serialization happens in router.js.
```
Comment (acorncom) [inline ember.js/march/march-03.md]: engine is *loaded*

File: ember.js/march/march-03.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+YK: And unfortunately the router map precompile step wasn't on the right path.
+
+#### Component Definition RFC
+
+GC: This is a WIP but I hope to wrap it up soon.
+
+YK: Constraint #1: you should never be required to re-enumerate all components just to specify what manager you want to use. It isn't good enough to say you only want to enumerate the glimmer ones, or only the ember ones.
+
+YK: Constraint 2: The information about which component manager you wnt to use should be require dynamically requiring the component file
```
Comment (acorncom) [inline ember.js/march/march-03.md]: you *want*

File: ember.js/march/march-03.md [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+YK: Constraint 3: all this needs to work with bare template components
+
+YK: I don't think in glimmer components JavaScript only component are a thing. In ember the JS-only would be needed for the root element, in Glimmer Components I cannot think of an instance where you want that.
+
+YK: In glimmer components there is no way to have anything in the DOM at all without a template. You can't have a div with out a template. This is not a crucial part of the point, but it is part of it.
+
+YK: I think what this means is that in the low leve of the system we need to have som emeta data that is.. I will start on the tempalte meta.  There it tells you which component you want to use. You can see if the tempalte meta has the component manager it knows which to call.
```
Comment (acorncom) [inline ember.js/march/march-03.md]: *low level*, *template* in multiple places

File: ember.js/march/march-03.md [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+
+GC: This is a WIP but I hope to wrap it up soon.
+
+YK: Constraint #1: you should never be required to re-enumerate all components just to specify what manager you want to use. It isn't good enough to say you only want to enumerate the glimmer ones, or only the ember ones.
+
+YK: Constraint 2: The information about which component manager you wnt to use should be require dynamically requiring the component file
+
+YK: Constraint 3: all this needs to work with bare template components
+
+YK: I don't think in glimmer components JavaScript only component are a thing. In ember the JS-only would be needed for the root element, in Glimmer Components I cannot think of an instance where you want that.
```
Comment (Turbo87) [inline ember.js/march/march-03.md]: @wycats one example might be https://github.com/miguelcobain/ember-leaflet

File: ember.js/march/march-03.md (hunk 1 above)
Comment (locks) [inline ember.js/march/march-03.md]: y u no PR :P

Comment (acorncom) [inline ember.js/march/march-03.md]: @locks didn't even occur to me! 😀

**Pull request merged**